pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -387,6 +391,79 @@ func (oe *operationExecutor) DetachVolume(
 	return oe.pendingOperations.Run(
 		volumeToDetach.VolumeName, "" /* podName */, detachFunc)
 }
+func (oe *operationExecutor) BulkVerifyVolumes(
+	attachedVolumes map[types.NodeName][]AttachedVolume,
+	actualStateOfWorld ActualStateOfWorldAttacherUpdater) error {
+
+	// A map of plugin names and nodes on which they exist with volumes they manage
+	bulkVerifyPluginsByNode := make(map[string]map[types.NodeName][]*volume.Spec)
+	volumeSpecMap := make(map[*volume.Spec]v1.UniqueVolumeName)
+
+	for node, nodeAttachedVolumes := range attachedVolumes {
+		for _, volumeAttached := range nodeAttachedVolumes {
+			volumePlugin, err :=
+				oe.operationGenerator.GetVolumePluginMgr().FindPluginBySpec(volumeAttached.VolumeSpec)
+
+			if err != nil || volumePlugin == nil {
+				glog.Errorf(
+					"VolumesAreAttached.FindPluginBySpec failed for volume %q (spec.Name: %q) on node %q with error: %v",
+					volumeAttached.VolumeName,
+					volumeAttached.VolumeSpec.Name(),
+					volumeAttached.NodeName,
+					err)
+				continue
+			}
+			attachableVolumePlugin, err :=
+				oe.operationGenerator.GetVolumePluginMgr().FindAttachablePluginByName(volumePlugin.GetPluginName())
+
+			if err != nil || attachableVolumePlugin == nil {
+				glog.Errorf(
+					"VolumeAreAttached.FindAttachablePluginBySpec failed for plugin %q with: %v",
+					volumePlugin.GetPluginName(),
+					err)
+				continue
+			}
+
+			volumeAttacher, newAttacherErr := attachableVolumePlugin.NewAttacher()
+
+			if newAttacherErr != nil {
+				glog.Errorf(
+					"VolumesAreAttached failed for getting plugin %q with: %v",
+					volumePlugin.GetPluginName(),
+					newAttacherErr)
+				continue
+			}
+
+			if _, ok := volumeAttacher.(volume.BulkVolumeVerifier); ok {
+				pluginNodes, pluginNodesExist := bulkVerifyPluginsByNode[volumePlugin.GetPluginName()]
+
+				if !pluginNodesExist {
+					pluginNodes = make(map[types.NodeName][]*volume.Spec)
+				}
+
+				volumeSpecList, nodeExists := pluginNodes[node]
+				if !nodeExists {
+					volumeSpecList = []*volume.Spec{}
+				}
+				volumeSpecList = append(volumeSpecList, volumeAttached.VolumeSpec)
+				pluginNodes[node] = volumeSpecList
+				bulkVerifyPluginsByNode[volumePlugin.GetPluginName()] = pluginNodes
+				volumeSpecMap[volumeAttached.VolumeSpec] = volumeAttached.VolumeName
+				continue
+			}
+			// If node doesn't support Bulk volume polling it is best to poll individually
+			oe.VerifyVolumesAreAttached(nodeAttachedVolumes, node, actualStateOfWorld)
+			break
+		}
+	}
+	bulkVerifyVolumeFunc, err :=
+		oe.operationGenerator.GenerateBulkVolumeVerifyFunc(bulkVerifyPluginsByNode, volumeSpecMap, actualStateOfWorld)
+
+	if err != nil {
+		return err
+	}
+	return oe.pendingOperations.Run("cluster-bulk-volume-poll", "" /* podName */, bulkVerifyVolumeFunc)
```

Comment:
"cluster-bulk-volume-poll" this is a fixed string. So it will prevent any concurrent VerifyVolumeFunc goroutine no matter what type of volume plugins?